utils/README.md

```diff
@@ -0,0 +1,150 @@
+# Utils
+
+
+## Prompt Client and Generation Tools
+
+This repository contains tools for generating prompts and interfacing with a Open AI API compliant vLLM-based inference API. The two main components are:
+
+1. `prompt_client_cli.py`: A command-line interface for sending prompts to a vLLM API server
+2. `prompt_generation.py`: A utility for generating and processing prompts with various configurations
+
+The primary usecase is stress testing of an Inference API server with many batches of inference with randomly varying context and timing.
+
+### Using prompt_client_cli.py
+
+The prompt client CLI tool allows you to send prompts to a vLLM API server with various configuration options.
+
+#### Environment Variables
+
+- `AUTHORIZATION`: Bearer token for API authentication
+- `JWT_SECRET`: Alternative to AUTHORIZATION for JWT-based authentication (is available in Docker container)
+- `DEPLOY_URL`: API server URL (default: http://127.0.0.1)
+- `SERVICE_PORT`: API server port (default: 8000)
+- `CACHE_ROOT`: Directory for saving response files (default: current directory)
+- `VLLM_MODEL`: Model name (default: meta-llama/Llama-3.1-70B-Instruct)
+
+#### Key Arguments
+
+- `--num_prompts`: Number of prompts to generate
+- `--batch_size`: Number of concurrent requests
+- `--max_prompt_length`: Maximum length for generated prompts
+- `--output_seq_len`: Maximum length for completions
+- `--num_full_iterations`: Number of times to repeat the full prompt set
+- `--vary-batch-size`: Randomize batch sizes using normal distribution
+- `--input_seq_len`: Fixed length for input sequences (-1 for variable)
+- `--inter_batch_delay`: Delay between batches in seconds
+- `--no-stream`: Disable streaming responses
+- `--dataset`: Source dataset (random, alpaca_eval)
+- `--distribution`: Prompt length distribution (fixed, uniform, normal)
```

Comment:
I'm not seeing `--distribution` under `prompt_client_cli.py`. 

Does `--distribution` need to be used with `--input_seq_len`? If `--input_seq_len` is set to a length such as 128, does that mean that `--distribution` must be set to `fixed`?

Reply:
Looks from here: https://github.com/tenstorrent/tt-inference-server/pull/41/files#diff-b27d6427aa7798c85e8aa50b17a6fd49d0a133be880921927eeb11a50232684fR306 that if a user sets `--input_seq_len -1` then it will just automatically get set to `--max_prompt_length`

Reply:
`--distribution` is only used in generate_random_prompts() which is called if `--dataset` is set to `"random"`. `--input_seq_len` in that case is the parameter for the distribution, i.e.`uniform: max`, `normal: mean, std`, `fixed -> all same length`.

Reply:
`max_prompt_length` is more important for when `--dataset = alpaca_eval`.

Reply:
Should we have checks for that? I don't want to complicate the code base, but it sounds like some combinations of configs wont be compatible together.

Reply:
They are compatible or disabled for the different usecases, e.g. for `--distribution` if the variable isnt needed it wont be used so the default is fine. The example usage shows the commands for expected usecases.